ts/src/main/scala/ch/epfl/ts/component/Component.scala

```diff
@@ -92,10 +92,27 @@ abstract class Component extends Receiver {
     case y if stopped => println("Received data when stopped " + this.getClass.getSimpleName + " of type " + y.getClass )
   }
 
-  // subclass can override do initialization here
+  /**
+   * Connects two compoenents
+   *
+   * Normally subclass don't need to override this method.
```

Comment:
Setting it `final` may be safer then?
We can always revert it later if we find a specific use case where it's required.